### ISSUE TITLE
Bug: Formatter disallows string data with application/json content-type.

### ISSUE THREAD

**Issue by trampgeek** [codeigniter4/CodeIgniter4#8397]
### PHP Version

8.1

### CodeIgniter4 Version

4.4.4

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

N/A

### What happened?

In a simple REST server controller, I call `$this->respond('Error message', 200)`.

The content type of the request is `application/json` and the response should be of the same type. But the format function in ResponseTrait.php considers passing a string as the `$data` parameter to be an error; it does not json_encode the string and changes the content type to `text/html`.

This seems wrong to me. As I read the spec for the application/json Mime type (see [RFC4627](https://datatracker.ietf.org/doc/html/rfc4627) and [RFC8259](https://datatracker.ietf.org/doc/html/rfc8259), the only requirement is that the response body be valid JSON, which includes integers, strings and bools, not just JSON objects. Thus the value of string in the response should have been json_encoded (which adds literal double-quote characters to each end) and the content type should have been left unchanged.

### Steps to Reproduce

Have a RESTful resource controller like:

```php
    namespace App\Controllers;

    use CodeIgniter\RESTful\ResourceController;
    use CodeIgniter\API\ResponseTrait;

    class Things extends ResourceController
    {
           public function post()
           {
                   return $this->respond('Error message', 200);
            }
    }
```

Route POST requests to the post method and send an HTTP POST request to the app, e.g. with a curl command like

    curl -H 'Content-Type: application/json' -d '{"x": "blah"}'  localhost:8080/things -v

The app returns with a content type of `text/html` and the response body is just

    Error message

The response should be `application/json` and the response body should be

    "Error message"

(where the double quotes should be present in the response).

### Expected Output

See above.

**Comment by kenjis** [comment]
Thank you for reporting.

@codeigniter4/core-team This seems to be a bug.

**Comment by michalsn** [comment]
Yes, and a fix sounds like a BC break 😜 

Although we had a similar conversation not so long ago when it comes to incoming JSON data. And we decided to not change anything if I remember correctly.

**Comment by kenjis** [comment]
I think we should follow the standard.

At least `getJSON()` can get all valid JSON data.

Enable Auto Routing Improved.
```php
<?php

namespace App\Controllers;

class Test extends BaseController
{
    public function postPost()
    {
        $json = $this->request->getJSON();

        return $this->response->setContentType('application/json')
            ->setBody(json_encode($json));
    }
}
```

```console
$ curl -H 'Content-Type: application/json' -d '100' localhost:8080/test/post 
100
```

```console
$ curl -H 'Content-Type: application/json' -d '"message"' localhost:8080/test/post 
"message"
```

```console
$ curl -H 'Content-Type: application/json' -d '{"foo": "bar"}' localhost:8080/test/post 
{"foo":"bar"}
```

**Comment by kenjis** [comment]
@trampgeek I sent a PR #8490

**Comment by trampgeek** [comment]
Many thanks for dealing with this, kenjis. Looks great. :-)

**Comment by kenjis** [comment]
@trampgeek If you believe the PR should be merged, approve it.
See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews

**Comment by trampgeek** [comment]
Ah, sorry. I wasn't familiar with the process. Thanks for clarifying. Have approved.

**Comment by kenjis** [comment]
Closed #8490
